tests/unit/test.h

```diff
@@ -55,4 +56,13 @@ void test_progress_end(void);
     }\
 }\
 
+// shared json test helpers, used in nova/unit/tests for cf-reactor
```

Comment:
@craigcomstock @larsewi @vpodzime Instead of exposing these in a `test.h`, how about just making them a part of the library? It could be a separate file `libntech/libutils/json_comparison.h`. Comparing JSONs could be usable in other places, as well.

Reply:
I'd expect us to already have that, actually.

Reply:
We do have
```C
int JsonCompare(const JsonElement *a, const JsonElement *b);
```
in `json.h`

Reply:
```C
    /* JsonCompare is naive and does not handle containers properly, thus we
     * need a custom comparison function to test for equality. */
```
I guess that is why there is a custom one in `json_test.c`? I have a feeling that I wrote this code once upon a time.

Reply:
I compared the JsonCompare in json.c with that in json_test.c and didn't see any functional difference so I guess we just don't need those and can use JsonCompare as-is. Tests pass here for me locally so will see how github likes it.